### PR TITLE
blog: docker build alternatives

### PR DESCRIPTION
**Pull request by nicks**
Hello @chimney42, @ellenkorbes,

Please review the following commits I made in branch nicks/imagebuilds:

f8f7de1466915974cf357235ded711a02ff55296 (2021-11-09 13:15:06 -0500)
blog: docker build alternatives

8fd3350a5aa2ce584c8836bae9b101842de0dfd4 (2021-11-04 15:33:55 -0400)
fix

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: blog/_posts/2021-11-04-docker-build-alternatives.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+[extensions](https://github.com/tilt-dev/tilt-extensions). The Tilt community
+frequently adds new types of image builders.
+
+In this post, I want to brag about three very different types of image builders that exist today:
+
+- `kubectl_build`,
+
+- `ko`, and
+
+- `buildpack`
```
Comment (nicks) [inline blog/_posts/2021-11-04-docker-build-alternatives.md:43]:
```suggestion
- `pack`
```

File: blog/_posts/2021-11-04-docker-build-alternatives.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+### Fun aside: Future Buildpack Work
+
+The layering approach to adding new functionality is getting a lot of excitement
+right now from container nerds.
+
+For security nerds, you can better track where source files come from and attach
+SBOMs (which are way beyond the scope of this post, but are fun to say out
+load.)
+
+One of the big reasons Tilt nerds are excited about buildpacks is that there's a
```
Comment (Chimney42) [inline blog/_posts/2021-11-04-docker-build-alternatives.md:207]:
```suggestion
One of the big reasons Tilters are excited about buildpacks is that there's a
```
I feel like you are using the term "nerds" a lot, you might want to switch it up once in a while 🤓

File: blog/_posts/2021-11-04-docker-build-alternatives.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
+---
+slug: "docker-build-alternatives"
+date: 2021-11-03
+author: nick
+layout: blog
+title: "How to Cope When Docker Builds are Slowing You Down"
```
Comment (Chimney42) [inline blog/_posts/2021-11-04-docker-build-alternatives.md:6]: I feel like "Docker builds slowing you down" restricts the problem space unnecessarily.
Instead, I'd suggest something like "When Docker Builds are not right for you" (or something better)

**Review by Chimney42 — APPROVED** [review]

File: blog/_posts/2021-11-04-docker-build-alternatives.md (hunk 3 above)
Comment (nicks) [reply blog/_posts/2021-11-04-docker-build-alternatives.md:6]: Ya, i was trying to do a hat-tip to Jason's tweet below. But I think it needs a catchier title. let me bat some ideas around

File: blog/_posts/2021-11-04-docker-build-alternatives.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+then save the results of those commands.
+
+The Ko view is: an image is a tarball. If you have a Mac, why do you need a
+Linux VM to run a table if you want a tarball? You already have good local tools
+for making tarballs.
+
+[Ko](https://github.com/google/ko#install) is an image builder targeted at Go
+apps. Go apps are particularly ill-suited for containers. The Go toolchain is
+well-optimized to run locally and builds a single binary. Ko builds your Go app
+locally, then packs it into an image.
```
[the remark this answers is not in the document]
Comment (nicks) [reply blog/_posts/2021-11-04-docker-build-alternatives.md]: done

File: blog/_posts/2021-11-04-docker-build-alternatives.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+- You can stack buildpacks to mix in the features you want (like hot-reload tools).
+
+- The buildpack keeps out lots of junk you don't want in the final image (like compilers).
+
+The `pack` extension was one of the first Tilt extensions! [Gareth
+Rushgrove](https://github.com/garethr) added it when we were still experimenting
+with the extension system.
+
+Since then, we worked closely with [Daniel
+Makusa](https://github.com/dmikusa-pivotal) of the buildpacks team to make it
```
Comment (dmikusa-pivotal) [inline blog/_posts/2021-11-04-docker-build-alternatives.md:171]:
```suggestion
Mikusa](https://github.com/dmikusa-pivotal) of the buildpacks team to make it
```

File: blog/_posts/2021-11-04-docker-build-alternatives.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+Read the full docs on the `ko` extension
+[here](https://github.com/tilt-dev/tilt-extensions/tree/master/ko).
+
+## Pack
+
+[`pack`](https://buildpacks.io/docs/tools/pack/) is also addressing people who
+are not happy maintaining Dockerfiles.
+
+But where Ko's philosophy states that you don't need a container to build an
+image, Buildpack takes the viewpoint that you need MORE containers.
```
Comment (dmikusa-pivotal) [inline blog/_posts/2021-11-04-docker-build-alternatives.md:152]: I would say the philosophy is that you shouldn't have to care how an image is built, like if a tool is using more or fewer containers to make that happen. Give `pack` your source code, it'll give you an image. What happens in between are details that you as an app dev don't need to manage or care about.

File: blog/_posts/2021-11-04-docker-build-alternatives.md (hunk 5 above)
Comment (nicks) [reply blog/_posts/2021-11-04-docker-build-alternatives.md:171]: oops, that's embarrassing, thank you

File: blog/_posts/2021-11-04-docker-build-alternatives.md (hunk 6 above)
Comment (nicks) [reply blog/_posts/2021-11-04-docker-build-alternatives.md:152]: haha, yes, this was a bit tongue-and-cheek, but you're right that i should fold in the actual philosophy and not the joke one

**Pull request merged**